compiler/internal/cueutil/build.go

```diff
@@ -0,0 +1,268 @@
+package cueutil
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"cuelang.org/go/cue"
+	"cuelang.org/go/cue/ast"
+	"cuelang.org/go/cue/build"
+	"cuelang.org/go/cue/cuecontext"
+	cueerrors "cuelang.org/go/cue/errors"
+	"cuelang.org/go/cue/load"
+	"cuelang.org/go/cue/parser"
+	"cuelang.org/go/cue/token"
+	"cuelang.org/go/encoding/json"
+	"golang.org/x/text/encoding/unicode"
+	"golang.org/x/text/transform"
+
+	"encr.dev/compiler/internal/cueutil/internal/yaml"
+	"encr.dev/pkg/eerror"
+)
+
+// LoadFromFS takes a given filesystem object and the app-relative path to the service's root package
+// and loads the full configuration needed for that service.
+func LoadFromFS(filesys fs.FS, serviceRelPath string) (cue.Value, error) {
+	// Write the FS to the file system
+	tmpPath, err := writeFSToTmpPath(filesys)
+	if err != nil {
+		return cue.Value{}, err
+	}
+	defer func() { _ = os.RemoveAll(tmpPath) }()
+
+	// Find all config files for the service
+	configFilesForService, err := allFilesUnder(filesys, serviceRelPath)
+	if err != nil {
+		return cue.Value{}, eerror.Wrap(err, "config", "unable to list all config files for service", map[string]any{"path": serviceRelPath})
+	}
+
+	// Tell CUE to load all the files
+	loaderCfg := &load.Config{
+		Dir:   tmpPath,
+		Tools: true,
+	}
+	pkgs := load.Instances(configFilesForService, loaderCfg)
+	for _, pkg := range pkgs {
+		if pkg.Err != nil {
+			return cue.Value{}, toDescriptiveError(pkg.Err)
+		}
+
+		// Non CUE files may be orphaned (JSON/YAML), so need to be parsed into the CUE AST and added to the package.
+		if err := addOrphanedFiles(pkg); err != nil {
+			return cue.Value{}, toDescriptiveError(err)
+		}
+	}
+
+	// Build the CUE values
+	ctx := cuecontext.New()
+	values, err := ctx.BuildInstances(pkgs)
+	if err != nil {
+		return cue.Value{}, toDescriptiveError(err)
+	}
+	if len(values) == 0 {
+		return cue.Value{}, eerror.New("config", "no values generated from config", nil)
+	}
+
+	// Unify all returned values into a single value
+	rtnValue := values[0]
+	for _, value := range values {
```

Comment:
When do we have multiple values? Is it one per package? (See above re: multiple packages)